### PR TITLE
Support DeviceID on Auxiliary Bus

### PR DESCRIPTION
**Pull request by adrianchiris**
Device plugins may allocate network device on a bus different than PCI.

sriov-network-device-plugin supports the allocation of network devices over Auxiliary bus[1][2][3].

extend host-device CNI to support such devices if provided through runtime config.

- Check if device provided by DeviceID runtime config is present on either PCI bus or Auxiliary bus
- extend getLink method to support getting netdev link obj from auxiliary bus
- add unit-test to cover the new flow

[1] https://github.com/k8snetworkplumbingwg/sriov-network-device-plugin/tree/master?tab=readme-ov-file#auxiliary-network-devices-selectors
[2] https://github.com/k8snetworkplumbingwg/sriov-network-device-plugin/tree/master/docs/subfunctions
[3] https://docs.kernel.org/networking/devlink/devlink-port.html

### CODE REVIEW

File: plugins/main/host-device/host-device.go [hunk 1]
```diff
@@ -53,6 +56,9 @@ type NetConf struct {
 	RuntimeConfig struct {
 		DeviceID string `json:"deviceID,omitempty"`
 	} `json:"runtimeConfig,omitempty"`
+
+	// for internal use
+	AuxDevice string `json:"-"` // Auxiliary device name as appears on Auxiliary bus (/sys/bus/auxiliary)
```
Comment (squeed) [inline plugins/main/host-device/host-device.go:61]: If this is internal, it the first word should be lower-cased. Yay golang.

Comment (adrianchiris) [reply plugins/main/host-device/host-device.go:61]: sure, will align.

noticed that `NetConf.DPDKMode` may also need to be internal as its inferred from the system[1].
but anyhow, not related to this PR.

[1] https://github.com/containernetworking/plugins/blob/d1aada912d434472ac3f709a1685cd9acfbb617a/plugins/main/host-device/host-device.go#L82

**Comment by adrianchiris** [comment]
@squeed golangci-lint version bumped in gh actions which causes lint failure (not related to this PR)

submitted #1009 to keep version fixed so that CI is stable (version bumps IMO should occur periodically in controlled manner, i.e submitting PR)

**Review by s1061123 — APPROVED** [review]
Looks good to me, thanks @adrianchiris !

**Comment by s1061123** [comment]
@squeed PTAL again?

**Comment by s1061123** [comment]
@adrianchiris hi, CI (Lint) has some error. Could you please take a look into it?

**Comment by adrianchiris** [comment]
> @adrianchiris hi, CI (Lint) has some error. Could you please take a look into it?

@s1061123  we should merge : #1009 then ill rebase this one and we should be good.

the lint issues not related to this PR but rather because linter version bumped in CI.

**Comment by adrianchiris** [comment]
Rebased on top of main. CI is green.

@s1061123 @s1061123 PTAL on this one :)

**Pull request merged**